packages/replay-next/components/errors/withSessionTimeoutCheck.ts

```diff
@@ -0,0 +1,30 @@
+import assert from "assert";
+import { ComponentType, createElement, useContext } from "react";
+import { ErrorBoundaryContext } from "react-error-boundary";
+
+import { ProtocolError, isCommandError } from "shared/utils/error";
+
+export function withSessionTimeoutCheck<Props extends object>(component: ComponentType<Props>) {
+  const Wrapper = (props: Props) => {
+    const context = useContext(ErrorBoundaryContext);
+    assert(context, "ErrorBoundaryContext is not available");
+
+    const { didCatch, error } = context;
+    if (didCatch) {
```

Comment:
Note to self: This won't handle the case where we show this inline fallback _manually_ (like in the source viewer)